pkg/admission/customresourcevalidation/clusterresourcequota/validate_crq.go

```diff
@@ -0,0 +1,161 @@
+package clusterresourcequota
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+
+	quotav1 "github.com/openshift/api/quota/v1"
+
+	"github.com/openshift/origin/pkg/admission/customresourcevalidation"
+	quotavalidation "github.com/openshift/origin/pkg/admission/customresourcevalidation/clusterresourcequota/validation"
+)
+
+const PluginName = "quota.openshift.io/ValidateClusterResourceQuota"
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(
+			map[schema.GroupResource]bool{
+				{Group: quotav1.GroupName, Resource: "clusterresourcequota"}:        true,
+				{Group: quotav1.GroupName, Resource: "appliedclusterresourcequota"}: true,
+			},
+			map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+				quotav1.GroupVersion.WithKind("ClusterResourceQuota"):        clusterResourceQuotaV1{},
+				quotav1.GroupVersion.WithKind("AppliedClusterResourceQuota"): appliedClusterResourceQuotaV1{},
+			})
+	})
+}
+
+func toClusterResourceQuota(uncastObj runtime.Object) (*quotav1.ClusterResourceQuota, field.ErrorList) {
+	if uncastObj == nil {
+		return nil, nil
+	}
+
+	allErrs := field.ErrorList{}
+
+	obj, ok := uncastObj.(*quotav1.ClusterResourceQuota)
+	if !ok {
+		return nil, append(allErrs,
+			field.NotSupported(field.NewPath("kind"), fmt.Sprintf("%T", uncastObj), []string{"ClusterResourceQuota"}),
+			field.NotSupported(field.NewPath("apiVersion"), fmt.Sprintf("%T", uncastObj), []string{"quota.openshift.io/v1"}))
+	}
+
+	return obj, nil
+}
+
+func toAppliedClusterResourceQuota(uncastObj runtime.Object) (*quotav1.AppliedClusterResourceQuota, field.ErrorList) {
```

Comment:
no.  appliedclusterresourcequota is a virtual resource, so it must stay aggregated.